## 01 Data_Excel-pd

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import random
from IPython.display import clear_output

### 1. Loading Data

#### Load CSV

In [2]:
par = "par1"
file = "par1_com2_data_2021-03-27-13.52.44"
path = "data/csv/{file}.csv".format(file = file)

Load CSV and remove timestamps

#### 1.1 Visual 1s

In [3]:
df_visual_1 = pd.read_csv(path)
df_visual_1 = df_visual_1.drop(["timestamps"], axis=1)
# df_visual_1.head()

/root/.local/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3072: DtypeWarning: Columns (17) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


#### 1.2 Visual 2s

In [4]:
df_visual_2 = pd.read_csv(path)
df_visual_2 = df_visual_2.drop(["timestamps"], axis=1)
# df_visual_1.head()

#### 1.3 Imagery 1s

In [5]:
df_imagery_1 = pd.read_csv(path)
df_imagery_1 = df_imagery_1.drop(["timestamps"], axis=1)
# df_imagery.head()

#### 1.4 Imagery 2s

In [6]:
df_imagery_2 = pd.read_csv(path)
df_imagery_2 = df_imagery_2.drop(["timestamps"], axis=1)
# df_imagery.head()

### 2. Checking Markers

Let's look at how the marker was generated.  Here is the format:

- [block, trial, label, time] <br>
label = 1,2,3,4,5,6,black <br>
time = 1s,2s

- [0] = warmup

In [7]:
print(df_visual_1['Marker'].unique())

['0' 'Training' '1,trial break,Fixation' '1,0,4,1' '1,task break,Fixation'
 '1,0,black,1' '1,0,1,2' '1,0,black,2' '1,0,5,2' '1,0,2,2' '1,0,6,2'
 '1,0,3,1' '1,1,4,2' '1,1,black,2' '1,1,1,2' '1,1,5,2' '1,1,2,1'
 '1,1,black,1' '1,1,6,2' '1,1,3,2' '1,2,4,1' '1,2,black,1' '1,2,1,1'
 '1,2,5,2' '1,2,black,2' '1,2,2,1' '1,2,6,1' '1,2,3,1' '1,3,4,1'
 '1,3,black,1' '1,3,1,1' '1,3,5,2' '1,3,black,2' '1,3,2,1' '1,3,6,1'
 '1,3,3,2' '1,4,4,1' '1,4,black,1' '1,4,1,2' '1,4,black,2' '1,4,5,1'
 '1,4,2,2' '1,4,6,1' '1,4,3,1' '2,trial break,Fixation' '2,0,4,1'
 '2,task break,Fixation' '2,0,black,1' '2,0,1,2' '2,0,black,2' '2,0,5,2'
 '2,0,2,2' '2,0,6,2' '2,0,3,2' '2,1,4,1' '2,1,black,1' '2,1,1,2'
 '2,1,black,2' '2,1,5,2' '2,1,2,1' '2,1,6,1' '2,1,3,1' '2,2,4,1'
 '2,2,black,1' '2,2,1,2' '2,2,black,2' '2,2,5,1' '2,2,2,2' '2,2,6,2'
 '2,2,3,1' '2,3,4,1' '2,3,black,1' '2,3,1,1' '2,3,5,1' '2,3,2,2'
 '2,3,black,2' '2,3,6,2' '2,3,3,2' '2,4,4,1' '2,4,black,1' '2,4,1,2'
 '2,4,black,2' '2,4,5,2' '2,4,2,2' '2,4,6,2' '2

Since we set our marker to have 4 info: #block, #trial, label, time.  We gonna split and get the class for the markers.   **Note that we shall reserve 0 for no event for raw mne, thus we shall represent class 0-9 using label 1-10.**

#### 2.1 Visual 1 s

In [8]:
#use numpy as another view of the pandas columns for faster operation
marker_np_visual_1 = df_visual_1['Marker'].values
marker_np_visual_1 = marker_np_visual_1.astype(str)

print(len(marker_np_visual_1))
### collecting indices of visual 1 s
visual_1_idx = []

for idx, marker in enumerate(marker_np_visual_1):
    if marker != '0':
        m = marker.split(",")
        if "Fixation" in marker:
            marker_np_visual_1[idx] = 0
        elif "trial break" in marker:
            marker_np_visual_1[idx] = 0
        elif "task break" in marker:
            marker_np_visual_1[idx] = 0
        elif "black" in marker: # remove black --> visual
            marker_np_visual_1[idx] = 0
        elif marker[-1] == '2':   # remove time = 2s --> 1s 
#             print('im HEREEEEEEEEEEEEEEEEEEEEE')
            marker_np_visual_1[idx] = 0
        elif "," in marker:
            marker_np_visual_1[idx] = m[2] # get classes
            visual_1_idx.append(idx)
        else:
            marker_np_visual_1[idx] = 0

# print(df_visual_1['Marker'].unique())
print(np.unique(marker_np_visual_1))
df_visual_1['Marker']= marker_np_visual_1

397380
['0' '1' '2' '3' '4' '5' '6']


In [9]:
print(df_visual_1.groupby('Marker').nunique())

            T3      T4      P4      P3      F4      F7      F3      F8  \
Marker                                                                   
0       177381  184309  201863  116909  182432  263091  254786  171227   
1           35      35      35      35      35      35      35      35   
2           37      37      37      37      37      37      37      37   
3           39      39      39      39      39      39      39      39   
4           35      35      35      35      35      35      35      35   
5           36      36      36      36      36      36      36      36   
6           39      39      39      39      39      39      39      39   

            O2      O1      T5      T6      C3      C4     FP1     FP2  
Marker                                                                  
0       139609  262757  147247  187322  231999  194774  217978  138447  
1           35      35      35      35      35      35      35      35  
2           37      37      37      37   

In [10]:
print('the number of stim in total: ',len(visual_1_idx))
print(visual_1_idx)

the number of stim in total:  221
[13886, 18258, 21670, 24188, 24874, 26438, 27118, 27784, 28451, 29117, 30719, 31375, 32946, 34532, 36127, 36791, 38693, 44005, 46506, 47213, 47902, 48564, 50120, 52589, 53256, 53919, 54594, 58049, 64549, 65217, 65907, 68410, 69074, 69722, 70384, 71079, 75464, 77058, 77731, 78400, 80899, 81569, 84084, 84742, 85426, 89192, 89855, 90527, 91209, 96489, 97181, 99701, 102226, 103790, 104439, 105113, 106741, 108340, 109013, 110633, 112187, 114096, 116569, 117263, 120668, 122244, 124716, 125397, 126081, 129546, 132999, 133676, 135281, 135945, 136634, 137281, 140118, 142614, 146096, 149533, 150203, 150857, 152442, 154028, 156489, 157181, 157829, 158482, 160081, 160749, 164492, 166061, 168650, 171245, 172886, 173583, 174267, 174933, 176520, 178148, 180712, 181376, 182103, 183706, 187147, 190018, 191610, 194174, 194867, 195525, 197138, 198732, 201404, 202121, 202809, 205326, 205980, 207620, 210177, 210869, 212501, 213154, 217879, 218550, 220176, 221786, 223357, 2

#### 2.2 Imagery 1 s

In [11]:
marker_np_imagery_1 = df_imagery_1['Marker'].values
marker_np_imagery_1 = marker_np_imagery_1.astype(str)

### collecting indices of imagery 1 s (the values of these indices will need to be replaced by the values of visual indices)
imagery_1_idx = [] # rows we need

for idx, marker in enumerate(marker_np_imagery_1):
    if marker != '0':
        m = marker.split(",")
        if "black" in marker and marker[-1] == '1' : # get black and 1s
            imagery_1_idx.append(idx)

In [12]:
print('the number of stim in total: ',len(imagery_1_idx))
print(imagery_1_idx)

the number of stim in total:  221
[14141, 18527, 21920, 24461, 25108, 26701, 27371, 28038, 28703, 29366, 30956, 31631, 33191, 34792, 36377, 37022, 38926, 44243, 46757, 47483, 48149, 48803, 50368, 52838, 53506, 54175, 54862, 58285, 64803, 65491, 66176, 68659, 69308, 69969, 70664, 71351, 75742, 77316, 77981, 78632, 81152, 81831, 84325, 85009, 85676, 89441, 90111, 90793, 91445, 96764, 97454, 99978, 102473, 104024, 104696, 105389, 106998, 108600, 109277, 110869, 112424, 114327, 116845, 117506, 120908, 122484, 124980, 125667, 126345, 129813, 133262, 133918, 135531, 136216, 136866, 137541, 140377, 142893, 146366, 149789, 150440, 151103, 152709, 154262, 156764, 157415, 158067, 158733, 160335, 161008, 164739, 166335, 168910, 171504, 173165, 173846, 174518, 175177, 176800, 178427, 180957, 181653, 182364, 183973, 187400, 190267, 191888, 194445, 195107, 195798, 197381, 198987, 201642, 202383, 203075, 205564, 206245, 207852, 210446, 211136, 212737, 213397, 218132, 218827, 220452, 222022, 223623, 2

### For example
df_imagery_1['Marker'][14141] = df_visual_1['Marker'].iloc[13886]

In [13]:
df_imagery_1['Marker'] = 0
for i in range(len(imagery_1_idx)):
    df_imagery_1['Marker'].iloc[imagery_1_idx[i]] = df_visual_1['Marker'].iloc[visual_1_idx[i]]

/usr/local/lib/python3.6/dist-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  iloc._setitem_with_indexer(indexer, value)


In [14]:
df_visual_1['Marker'].iloc[visual_1_idx[0]]

'4'

In [15]:
print(df_imagery_1.groupby('Marker').nunique())

            T3      T4      P4      P3      F4      F7      F3      F8  \
Marker                                                                   
0       177375  184315  201863  116901  182438  263114  254787  171223   
1           35      35      35      35      35      35      35      35   
2           37      37      37      37      37      37      37      37   
3           39      39      39      39      39      39      39      39   
4           35      35      35      35      35      35      35      35   
5           36      36      36      36      36      36      36      36   
6           39      39      39      39      39      39      39      39   

            O2      O1      T5      T6      C3      C4     FP1     FP2  
Marker                                                                  
0       139613  262762  147245  187307  231992  194782  217983  138453  
1           35      35      35      35      35      35      35      35  
2           37      37      37      37   

#### 2.3 Visual 2 s

In [16]:
#use numpy as another view of the pandas columns for faster operation
marker_np_visual_2 = df_visual_2['Marker'].values
marker_np_visual_2 = marker_np_visual_2.astype(str)

### collecting indices of visual 2 s
visual_2_idx = []

for idx, marker in enumerate(marker_np_visual_2):
    if marker != '0':
        m = marker.split(",")
#         print(m[-1])
        if "Fixation" in marker:
            marker_np_visual_2[idx] = 0
        elif "trial break" in marker:
            marker_np_visual_2[idx] = 0
        elif "task break" in marker:
            marker_np_visual_2[idx] = 0
        elif "black" in marker: # remove black --> visual
            marker_np_visual_2[idx] = 0
        elif marker[-1] == '1':   # remove time = 1s --> 2s 
#             print('im HEREEEEEEEEEEEEEEEEEEEEE')
            marker_np_visual_2[idx] = 0
        elif "," in marker:
            marker_np_visual_2[idx] = m[2] # get classes
            visual_2_idx.append(idx)
        else:
            marker_np_visual_2[idx] = 0

# print(df_visual_1['Marker'].unique())
print(np.unique(marker_np_visual_2))
df_visual_2['Marker']= marker_np_visual_2

['0' '1' '2' '3' '4' '5' '6']


In [17]:
print('the number of stim in total: ',len(visual_2_idx))
print(visual_2_idx)

the number of stim in total:  229
[14556, 15495, 16403, 17335, 18940, 19846, 20765, 22335, 23260, 25525, 29783, 32046, 33604, 35207, 39339, 40280, 41218, 42154, 43075, 44656, 45574, 49218, 50782, 51686, 55277, 56188, 57121, 58700, 59626, 60527, 61465, 62386, 66591, 67505, 71766, 72673, 73616, 74523, 76158, 79045, 79968, 82247, 83150, 86093, 87023, 91863, 92771, 93707, 94613, 95545, 97869, 98795, 100397, 101299, 102891, 105803, 107414, 109694, 111286, 114744, 115660, 117921, 118842, 119764, 121323, 122899, 123797, 126766, 127695, 128614, 130234, 131160, 132092, 134332, 139213, 140795, 141700, 143309, 144257, 145155, 146782, 147728, 148623, 151518, 153124, 154676, 155589, 159147, 161424, 162329, 165154, 166750, 167669, 169347, 170283, 171952, 175594, 177216, 178844, 179774, 182779, 184388, 185299, 186240, 187815, 190695, 192307, 193240, 196222, 197803, 199407, 200411, 203491, 204419, 206677, 208267, 209244, 211581, 215096, 216011, 216939, 219244, 220877, 222450, 224729, 225648, 229938, 2

In [18]:
print(df_visual_2.groupby('Marker').nunique())

            T3      T4      P4      P3      F4      F7      F3      F8  \
Marker                                                                   
0       177381  184313  201848  116910  182440  263117  254789  171237   
1           40      40      40      40      40      40      40      40   
2           38      38      38      38      38      38      38      38   
3           36      36      36      36      36      36      36      36   
4           40      40      40      40      40      40      40      40   
5           39      39      39      39      39      39      39      39   
6           36      36      36      36      36      36      36      36   

            O2      O1      T5      T6      C3      C4     FP1     FP2  
Marker                                                                  
0       139610  262754  147249  187322  231985  194769  217972  138450  
1           40      40      40      40      40      40      40      40  
2           38      38      38      38   

#### 2.4 Imagery 2 s

In [19]:
marker_np_imagery_2 = df_imagery_2['Marker'].values
marker_np_imagery_2 = marker_np_imagery_2.astype(str)

### collecting indices of imagery 2 s (the values of these indices will need to be replaced by the values of visual indices)
imagery_2_idx = []
for idx, marker in enumerate(marker_np_imagery_2):
    if marker != '0':
        m = marker.split(",")
        if "black" in marker and marker[-1] == '2' : # get black and 2s
            imagery_2_idx.append(idx)


In [20]:
print('the number of stim in total: ',len(imagery_2_idx))
print(imagery_2_idx)

the number of stim in total:  229
[14953, 15863, 16794, 17720, 19309, 20223, 21132, 22721, 23648, 25897, 30179, 32407, 33992, 35586, 39739, 40680, 41613, 42537, 43465, 45033, 45967, 49579, 51145, 52050, 55645, 56582, 57507, 59086, 59987, 60923, 61847, 62753, 66964, 67871, 72133, 73076, 73985, 74921, 76517, 79426, 80358, 82610, 83545, 86482, 87394, 92230, 93166, 94073, 95003, 95946, 98254, 99162, 100759, 101683, 103249, 106198, 107799, 110092, 111644, 115117, 116030, 118302, 119224, 120125, 121697, 123257, 124177, 127155, 128074, 129007, 130613, 131550, 132458, 134739, 139580, 141161, 142074, 143711, 144616, 145555, 147186, 148084, 148991, 151900, 153487, 155048, 155949, 159539, 161787, 162695, 165517, 167124, 168067, 169743, 170678, 172341, 175980, 177609, 179229, 180161, 183167, 184758, 185700, 186603, 188210, 191058, 192694, 193629, 196589, 198185, 199775, 200801, 203877, 204787, 207076, 208673, 209633, 211959, 215470, 216397, 217341, 219634, 221246, 222810, 225104, 226032, 230338, 2

In [21]:
df_imagery_2['Marker'] = '0'
for i in range(len(imagery_2_idx)):
    df_imagery_2['Marker'].iloc[imagery_2_idx[i]] = df_visual_2['Marker'].iloc[visual_2_idx[i]]

/usr/local/lib/python3.6/dist-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  iloc._setitem_with_indexer(indexer, value)


In [22]:
print(df_imagery_2.groupby('Marker').nunique())

            T3      T4      P4      P3      F4      F7      F3      F8  \
Marker                                                                   
0       177388  184310  201862  116900  182436  263096  254782  171227   
1           40      40      40      40      40      40      40      40   
2           38      38      38      38      38      38      38      38   
3           36      36      36      36      36      36      36      36   
4           40      40      40      40      40      40      40      40   
5           39      39      39      39      39      39      39      39   
6           36      36      36      36      36      36      36      36   

            O2      O1      T5      T6      C3      C4     FP1     FP2  
Marker                                                                  
0       139604  262763  147237  187312  231987  194763  217973  138453  
1           40      40      40      40      40      40      40      40  
2           38      38      38      38   

In [23]:
### color  ####

color = { 'green': ['4','5'], 'red':['1','3'] , 'blue':['2','6'] }

### shape ####
shape = { 'cone':['2','3','4'] ,'cube': ['1','5','6'] }


## time ###

In [24]:
df_visual_1['Marker'].unique()

array(['0', '4', '3', '2', '1', '6', '5'], dtype=object)

### 2.1.1 Visual 1 s and green 

In [25]:
df_visual_1_green = df_visual_1.copy()

In [26]:
#use numpy as another view of the pandas columns for faster operation
marker_visual_1_green = df_visual_1_green['Marker'].values
marker_visual_1_green = marker_visual_1_green.astype(str)

### collecting
visual_1_green_idx = []

In [27]:
for idx, marker in enumerate(marker_visual_1_green):
    if marker != '0':
#         print(m[-1])
        if color['green'][0] in marker or color['green'][1] in marker:
             #marker_visual_1_green[idx] = m[2] # get classes
             visual_1_green_idx.append(idx)
        else:
            df_visual_1_green['Marker'][idx] = '0'
        
            

# print(df_visual_1['Marker'].unique())
# print(np.unique(marker_np_visual_2))
# df_visual_2['Marker']= marker_np_visual_2

/root/.local/lib/python3.6/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [28]:
print(len(visual_1_green_idx)) 

71


In [29]:
df_visual_1_green['Marker'].unique()

array(['0', '4', '5'], dtype=object)

In [30]:
#visual_1_green_idx

In [31]:
df_visual_1['Marker'][visual_1_green_idx].unique()

array(['4', '5'], dtype=object)

### 2.1.2 Visual 1 s and red 

In [32]:
df_visual_1['Marker'].unique()

array(['0', '4', '3', '2', '1', '6', '5'], dtype=object)

In [33]:
df_visual_1_red = df_visual_1.copy()

In [34]:
df_visual_1_red['Marker'].unique()

array(['0', '4', '3', '2', '1', '6', '5'], dtype=object)

In [35]:
#use numpy as another view of the pandas columns for faster operation
marker_visual_1_red = df_visual_1_red['Marker'].values
marker_visual_1_red = marker_visual_1_red.astype(str)

### collecting
visual_1_red_idx = [] 

In [36]:
#df_imagery_2

In [37]:
for idx, marker in enumerate(marker_visual_1_red):
    if marker != '0':
#         print(m[-1])
        if color['red'][0] in marker or color['red'][1] in marker:
             #marker_visual_1_green[idx] = m[2] # get classes
             visual_1_red_idx.append(idx)
        else:
            df_visual_1_red['Marker'][idx] = '0'
            
# print(df_visual_1['Marker'].unique())
# print(np.unique(marker_np_visual_2))
# df_visual_2['Marker']= marker_np_visual_2

/root/.local/lib/python3.6/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [38]:
print(len(visual_1_red_idx)) 

74


In [39]:
df_visual_1_red['Marker'].unique()

array(['0', '3', '1'], dtype=object)

In [40]:
#visual_1_green_idx

In [41]:
df_visual_1['Marker'][visual_1_red_idx].unique()

array(['3', '1'], dtype=object)

### 2.1.3 Visual 1 s and blue

In [42]:
df_visual_1_blue = df_visual_1.copy()

In [43]:
df_visual_1_blue['Marker'].unique()

array(['0', '4', '3', '2', '1', '6', '5'], dtype=object)

In [44]:
#use numpy as another view of the pandas columns for faster operation
marker_visual_1_blue = df_visual_1_blue['Marker'].values
marker_visual_1_blue = marker_visual_1_blue.astype(str)

### collecting
visual_1_blue_idx = []

In [45]:
for idx, marker in enumerate(marker_visual_1_blue):
    if marker != '0':
#         print(m[-1])
        if color['blue'][0] in marker or color['blue'][1] in marker:
             #marker_visual_1_green[idx] = m[2] # get classes
             visual_1_blue_idx.append(idx)
        else:
            df_visual_1_blue['Marker'][idx] = '0'
        
            

# print(df_visual_1['Marker'].unique())
# print(np.unique(marker_np_visual_2))
# df_visual_2['Marker']= marker_np_visual_2

/root/.local/lib/python3.6/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [46]:
print(len(visual_1_blue_idx)) 

76


In [47]:
df_visual_1_blue['Marker'].unique()

array(['0', '2', '6'], dtype=object)

In [48]:
#visual_1_green_idx

In [49]:
df_visual_1['Marker'][visual_1_blue_idx].unique()

array(['2', '6'], dtype=object)

### 2.1.4 Visual 1 s and Cone

In [50]:
df_visual_1_cone = df_visual_1.copy()

In [51]:
df_visual_1_cone['Marker'].unique()

array(['0', '4', '3', '2', '1', '6', '5'], dtype=object)

In [52]:
#use numpy as another view of the pandas columns for faster operation
marker_visual_1_cone = df_visual_1_cone['Marker'].values
marker_visual_1_cone = marker_visual_1_cone.astype(str)

### collecting
visual_1_cone_idx = []

In [53]:
for idx, marker in enumerate(marker_visual_1_cone):
    if marker != '0':
#         print(m[-1])
        if shape['cone'][0] in marker or shape['cone'][1] in marker or shape['cone'][2] in marker:
             #marker_visual_1_green[idx] = m[2] # get classes
             visual_1_cone_idx.append(idx)
        
        else:
            df_visual_1_cone['Marker'][idx] = '0'

# print(df_visual_1['Marker'].unique())
# print(np.unique(marker_np_visual_2))
# df_visual_2['Marker']= marker_np_visual_2

/root/.local/lib/python3.6/site-packages/ipykernel_launcher.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if __name__ == '__main__':


In [54]:
print(len(visual_1_cone_idx)) 

111


In [55]:
#visual_1_green_idx

In [56]:
df_visual_1_cone['Marker'][visual_1_cone_idx].unique()

array(['4', '3', '2'], dtype=object)

In [57]:
df_visual_1_cone['Marker'].unique()

array(['0', '4', '3', '2'], dtype=object)

### 2.1.4 Visual 1 s and Cube

In [58]:
df_visual_1_cube = df_visual_1.copy()

In [59]:
#use numpy as another view of the pandas columns for faster operation
marker_visual_1_cube = df_visual_1_cube['Marker'].values
marker_visual_1_cube = marker_visual_1_cube.astype(str)

### collecting
visual_1_cube_idx = []

In [60]:
shape['cone'][2]

'4'

In [61]:
for idx, marker in enumerate(marker_visual_1_cube):
    if marker != '0':
#         print(m[-1])
        if shape['cube'][0] in marker or shape['cube'][1] in marker or shape['cube'][2] in marker:
             #marker_visual_1_green[idx] = m[2] # get classes
             visual_1_cube_idx.append(idx)
        else:
            df_visual_1_cube['Marker'][idx] = '0'
        
            

# print(df_visual_1['Marker'].unique())
# print(np.unique(marker_np_visual_2))
# df_visual_2['Marker']= marker_np_visual_2

/root/.local/lib/python3.6/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [62]:
print(len(visual_1_cube_idx)) 

110


In [64]:
#visual_1_green_idx

In [65]:
df_visual_1['Marker'][visual_1_cube_idx].unique()

array(['1', '6', '5'], dtype=object)

In [66]:
df_visual_1_cube['Marker'].unique()

array(['0', '1', '6', '5'], dtype=object)

In [72]:
df_visual_2['Marker'].unique()

array(['0', '1', '5', '2', '6', '4', '3'], dtype=object)

In [71]:
df_visual_1_red['Marker'].unique()  , df_visual_1_green['Marker'].unique() , df_visual_1_blue['Marker'].unique()  

(array(['0', '3', '1'], dtype=object),
 array(['0', '4', '5'], dtype=object),
 array(['0', '2', '6'], dtype=object))

In [73]:
df_visual_1_cube['Marker'].unique() , df_visual_1_cone['Marker'].unique() 

(array(['0', '1', '6', '5'], dtype=object),
 array(['0', '4', '3', '2'], dtype=object))

### 3. Save Data

In [74]:
# df_visual_1_red.to_pickle("data/pd/{par}/{file}_visual_1_red.pkl".format(par=par, file=file))
# df_visual_1_green.to_pickle("data/pd/{par}/{file}_visual_1_green.pkl".format(par=par, file=file))
# df_visual_1_blue.to_pickle("data/pd/{par}/{file}_visual_1_blue.pkl".format(par=par, file=file))
# df_visual_1_cube.to_pickle("data/pd/{par}/{file}_visual_1_cube.pkl".format(par=par, file=file))
# df_visual_1_cone.to_pickle("data/pd/{par}/{file}_visual_1_cone.pkl".format(par=par, file=file))

In [ ]:
#df_visual_1.to_pickle("data/pd/{par}/{file}_visual_1.pkl".format(par=par, file=file))

In [67]:
df_visual_1.to_pickle("data/pd/{par}/{file}_visual_1.pkl".format(par=par, file=file))
# df_imagery_1.to_pickle("data/pd/{par}/{file}_imagery_1.pkl".format(par=par, file=file))
# df_visual_2.to_pickle("data/pd/{par}/{file}_visual_2.pkl".format(par=par, file=file))
# df_imagery_2.to_pickle("data/pd/{par}/{file}_imagery_2.pkl".format(par=par, file=file))